leopard.go

```
@@ -1171,6 +1265,28 @@ func initMul16LUT() {
 	}
 }
 
+func (r *leopardFF16) getShardSlice() [][]byte {
+	if s, ok := r.shardSlicePool.Get().([][]byte); ok {
+		return s[:r.totalShards]
+	}
+	return make([][]byte, r.totalShards)
+}
+
+func (r *leopardFF16) putShardSlice(s [][]byte) {
+	r.shardSlicePool.Put(s)
```

golangci-lint (2.11.4)

[error] SA6002: argument should be pointer-like to avoid allocations

(staticcheck)